backend-rust/src/indexer.rs

```diff
@@ -1919,10 +2132,181 @@ impl PreparedContractUpdate {
         .execute(tx.as_mut())
         .await?;
 
-        Ok(())
+        process_cis2_events(
+            &self.cis2_events,
+            self.contract_index,
+            self.contract_sub_index,
+            pool,
+            transaction_index,
+            tx,
+        )
+        .await
     }
 }
 
+async fn process_cis2_events(
+    cis2_events: &[cis2::Event],
+    contract_index: i64,
+    contract_sub_index: i64,
+    pool: &PgPool,
+    transaction_index: i64,
+    tx: &mut sqlx::Transaction<'static, sqlx::Postgres>,
+) -> anyhow::Result<()> {
+    for log in cis2_events.iter() {
+        // The `total_supply` value of a token is inserted/updated in the database here.
+        // Only `Mint` and `Burn` events affect the `total_supply` of a
+        // token.
+        if let cis2::Event::Mint {
```

Comment:
Use `match` statement instead of several `if lets` as this convey better the control flow and check for exhaustive matches